packages/@aws-cdk/core/lib/stack.ts

```diff
@@ -1629,7 +1633,7 @@ export function rootPathTo(construct: IConstruct, ancestor?: IConstruct): IConst
  * behavior.
  */
 function makeStackName(components: string[]) {
-  if (components.length === 1) { return components[0]; }
+  if (components.length === 1 && components[0].length <= 128) { return components[0]; }
```

Comment:
Iirc, this check is done inside `makeUniqueResourceName` so it shouldn't be needed here.

Reply:
The check done inside `makeUniqueResourceName` will remove the `-` from the prefix, as it removes anything that is not a number or a character, unless you specify to keep a special character though parameters, but I was afraid of breaking compatibility with already existing stacks if doing so.
I'll make changes to pass the prefix to the `makeUniqueResourceName` as an optional parameter as you mentioned in another comment.
Thanks!